src/_pytest/nodes.py

```diff
@@ -462,6 +462,9 @@ def reportinfo(self) -> Tuple[Union[py.path.local, str], Optional[int], str]:
     @cached_property
     def location(self) -> Tuple[str, Optional[int], str]:
         location = self.reportinfo()
-        fspath = self.session._node_location_to_relpath(location[0])
+        fspath = location[0]
```

Comment:
I prefer writing it as

```py
        if isinstance(location[0], py.path.local):
            fspath = location[0]
        else:
            fspath = py.path.local(location[0])
        relfspath = self.session._node_location_to_relpath(fspath)
```

This way there is no type confusion.

Reply:
:+1: Amended.